packages/utils/test/schema/toPathSchemaTest.ts

```diff
@@ -702,5 +702,179 @@ export default function toPathSchemaTest(testValidator: TestValidatorType) {
+      });
+    });
+
+    it('should handle fixed array items which are objects, which have additionalItems in the schema and have additional items in the form data', () => {
+      const schema: RJSFSchema = {
+        type: 'object',
+        properties: {
+          str: { type: 'string' },
+          arr: {
+            type: 'array',
+            items: [
+              {
+                type: 'object',
+                properties: {
+                  name: { type: 'string' },
+                },
+              },
+            ],
+            additionalItems: {
+              type: 'string',
+            },
+          },
+        },
+      };
+
+      const formData = {
+        str: 'string',
+        arr: [{ name: 'name1' }, 'name2'],
+      };
+
+      expect(toPathSchema(testValidator, schema, '', schema, formData)).toEqual({
+        $name: '',
+        arr: {
+          $name: 'arr',
+          '0': {
+            $name: 'arr.0',
+            name: {
+              $name: 'arr.0.name',
+            },
+          },
+          '1': {
+            $name: 'arr.1',
+          },
+        },
+        str: {
+          $name: 'str',
+        },
+      });
+    });
+
+    it("should handle fixed array items which are objects, which have additionalItems in the schema but don't have additional items in the form data", () => {
+      const schema: RJSFSchema = {
+        type: 'object',
+        properties: {
+          str: { type: 'string' },
+          arr: {
+            type: 'array',
+            items: [
+              {
+                type: 'object',
+                properties: {
+                  name: { type: 'string' },
+                },
+              },
+            ],
+            additionalItems: {
+              type: 'string',
+            },
+          },
+        },
+      };
+
+      const formData = {
+        str: 'string',
+        arr: [{ name: 'name1' }],
+      };
+
+      expect(toPathSchema(testValidator, schema, '', schema, formData)).toEqual({
+        $name: '',
+        arr: {
+          $name: 'arr',
+          '0': {
+            $name: 'arr.0',
+            name: {
+              $name: 'arr.0.name',
+            },
+          },
+        },
+        str: {
+          $name: 'str',
+        },
+      });
+    });
+
+    it('should handle fixed array items which are objects, which have additionalItems as false in the schema and have additional items in the form data', () => {
+      const schema: RJSFSchema = {
+        type: 'object',
+        properties: {
+          str: { type: 'string' },
+          arr: {
+            type: 'array',
+            items: [
+              {
+                type: 'object',
+                properties: {
+                  name: { type: 'string' },
+                },
+              },
+            ],
+            additionalItems: false,
+          },
+        },
+      };
+
+      const formData = {
+        str: 'string',
+        arr: [{ name: 'name1' }, 'name2'],
```

Comment:
I'm wondering whether there should be a `console.warn` or `console.error` for this condition?

Reply:
I can see that `toPathSchema` util is mostly used as a step to omit extra data. In that case should we warn the users about the omitted data?